joint_state_controller:
  type: "joint_state_controller/JointStateController"
  publish_rate: 50  

base_controller:
  type        : "diff_drive_controller/DiffDriveController"
  left_wheel  : 'l_wheel_joint'
  right_wheel : 'r_wheel_joint'
  publish_rate: 50.0               # default: 50
  pose_covariance_diagonal: [0.001, 0.001, 0.001, 0.001, 0.001, 0.03]
  twist_covariance_diagonal: [0.001, 0.001, 0.001, 0.001, 0.001, 0.03]
  wheel_radius : 0.0615
  enable_odom_tf : false
  linear:
    x:
      has_velocity_limits    : true
      max_velocity           : 0.75  # m/s
      min_velocity           : -0.5 # m/s
      has_acceleration_limits: true
      max_acceleration       : 1  # m/s^2
      min_acceleration       : -1 # m/s^2
  angular:
    z:
      has_velocity_limits    : true
      max_velocity           : 1.5  # rad/s
      has_acceleration_limits: true
      max_acceleration       : 2  # rad/s^2


arm_controller:
  type: "effort_controllers/JointTrajectoryController"
  joints:
    - shoulder_pan_joint
    - shoulder_lift_joint
    - upperarm_roll_joint
    - elbow_flex_joint
    - forearm_roll_joint
    - wrist_flex_joint
    - wrist_roll_joint

  constraints:
    goal_time: 1.0                  # Override default
    shoulder_pan_joint: {trajectory: 0.2, goal: 0.1}
    shoulder_lift_joint: {trajectory: 0.2, goal: 0.1}
    upperarm_roll_joint: {trajectory: 0.2, goal: 0.1}
    elbow_flex_joint: {trajectory: 0.2, goal: 0.1}
    forearm_roll_joint: {trajectory: 0.2, goal: 0.1}
    wrist_flex_joint: {trajectory: 0.2, goal: 0.1}
    wrist_roll_joint: {trajectory: 0.2, goal: 0.1}
  
  gains:
    shoulder_pan_joint: {p: 2000,  d: 10, i: 0, i_clamp: 0}
    shoulder_lift_joint: {p: 2000,  d: 10, i: 0, i_clamp: 0}
    upperarm_roll_joint: {p: 2000,  d: 10, i: 0, i_clamp: 0}
    elbow_flex_joint: {p: 2000,  d: 10, i: 0, i_clamp: 0}
    forearm_roll_joint: {p: 2000,  d: 10, i: 0, i_clamp: 0}
    wrist_flex_joint: {p: 2000,  d: 10, i: 0, i_clamp: 0}
    wrist_roll_joint: {p: 2000,  d: 10, i: 0, i_clamp: 0}

head_controller:
  type: "effort_controllers/JointTrajectoryController"
  joints:
    - head_pan_joint
    - head_tilt_joint
  gains:
    head_pan_joint: {p: 2000,  d: 10, i: 0, i_clamp: 0}  
    head_tilt_joint: {p: 2000,  d: 10, i: 0, i_clamp: 0}  

# gripper_controller:
#   type: "effort_controllers/JointTrajectoryController"
#   joints:
#     - r_gripper_finger_joint
#     - l_gripper_finger_joint
#   gains:
#     r_gripper_finger_joint: {p: 2000,  d: 10, i: 0, i_clamp: 0}  
#     l_gripper_finger_joint: {p: 2000,  d: 10, i: 0, i_clamp: 0}        

torso_lift_controller:
  type: effort_controllers/JointPositionController
  joint: torso_lift_joint
  pid: {p: 1000.0, i: 0.0, d: 0.0}

r_gripper_finger_controller:
  type: effort_controllers/JointPositionController
  joint: r_gripper_finger_joint
  pid: {p: 5000.0, i: 0.0, d: 0.0}

l_gripper_finger_controller:
  type: effort_controllers/JointPositionController
  joint: l_gripper_finger_joint
  pid: {p: 5000.0, i: 0.0, d: 0.0}

bellows_controller:
  type: effort_controllers/JointPositionController
  joint: bellows_joint
  pid: {p: 5000.0, i: 0.0, d: 0.0}


